framework/src/actions/CopyNodalVarsAction.C

```diff
@@ -57,11 +57,26 @@ CopyNodalVarsAction::act()
     else
     {
       SystemBase * system;
+      SystemBase * tempSys;
       // Is this a NonlinearSystem variable or an AuxiliarySystem variable?
       if (_current_task == "copy_nodal_vars")
+      {
+        // This iterates through each nonlinear system and finds which one the current variable
+        // needs to be copied to
         system = &_problem->getNonlinearSystemBase(/*nl_sys=*/0);
```

Comment:
no need to set this if you are going to loop after to find the real one

Reply:
My thought with this was that it should be set to the first nonlinear system by default. That way if it is in fact a typo and can't be assigned to any of them, then when it tries to copy later it should throw an error instead of not trying to copy it. If it's better though I can have it throw an error right there it it isn't able to assign the variable to any of the nonlinear systems?

Reply:
that works

Reply:
I did some testing and it looks like assigning a variable by default to the first nonlinear system if unnecessary. If the user does have typos in their variable names will cause errors elsewhere so having a check there to make sure it was assigned doesn't do anything, so I made the changes you recommended.